modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_2/test/ObjectDefinitionUpgradeProcessTest.java

```diff
@@ -0,0 +1,121 @@
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.object.definition.util.ObjectDefinitionUtil;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.persistence.BasePersistence;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Tavares
+ */
+@RunWith(Arquillian.class)
+public class ObjectDefinitionUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@After
+	public void tearDown() throws Exception {
+		_companyLocalService.forEachCompany(
+			company -> {
+				for (ObjectDefinition objectDefinition :
+						_objectDefinitionLocalService.
+							getUnmodifiableSystemObjectDefinitions(
+								company.getCompanyId())) {
+
+					String externalReferenceCode =
+						ObjectDefinitionUtil.
+							getUnmodifiableSystemObjectDefinitionExternalReferenceCode(
+								objectDefinition.getName());
+
+					if (StringUtil.equals(
+							externalReferenceCode,
+							objectDefinition.getExternalReferenceCode())) {
+
+						continue;
+					}
+
+					_objectDefinitionLocalService.updateExternalReferenceCode(
+						objectDefinition.getObjectDefinitionId(),
+						externalReferenceCode);
+				}
+			});
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		for (ObjectDefinition objectDefinition :
+				_objectDefinitionLocalService.
+					getUnmodifiableSystemObjectDefinitions(
+						TestPropsValues.getCompanyId())) {
+
+			_objectDefinitionLocalService.updateExternalReferenceCode(
+				objectDefinition.getObjectDefinitionId(),
+				RandomTestUtil.randomString());
+		}
+
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator,
+			"com.liferay.object.internal.upgrade.v9_2_2." +
+				"ObjectDefinitionUpgradeProcess");
+
+		upgradeProcess.upgrade();
+
+		BasePersistence<?> basePersistence =
+			_objectDefinitionLocalService.getBasePersistence();
+
+		basePersistence.clearCache();
```

Comment:
@devpedrotavares I'm not sure if there is a significant difference between this clear cache approach and `_multiVMPool.clear();`, but I'd like you replace this one with `_multiVMPool.clear();` because most of UpgradeProcessTest classes are following this pattern, so I believe we should keep following it, unless we have a specific reason to use the basePersistence.clearCache. [see](https://github.com/liferay/liferay-portal/blob/4816522526227d8802b356f1efeee240911bae38/modules/apps/portal-security/portal-security-script-management-test/src/testIntegration/java/com/liferay/portal/security/script/management/web/internal/upgrade/v1_1_0/test/ScriptManagementConfigurationUpgradeProcessTest.java#L265-L276)

Thanks!

Reply:
Done!